restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverterService.java

```diff
@@ -0,0 +1,290 @@
+package io.stargate.web.docsapi.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.*;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.schema.Column;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+public class JsonConverterService {
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  public ObjectMapper getMapper() {
+    return mapper;
+  }
+
+  // TODO find a place to refactor this to, along with DocumentService#getBooleanFromRow
+  private static Boolean getBooleanFromRow(Row row, String colName, boolean numericBooleans) {
+    if (row.isNull("bool_value")) return null;
+    if (numericBooleans) {
+      byte value = row.getByte(colName);
+      return value != 0;
+    }
+    return row.getBoolean(colName);
+  }
+
+  public ImmutablePair<JsonNode, Map<String, List<JsonNode>>> convertToJsonDoc(
+      List<Row> rows, boolean writeAllPathsAsObjects, boolean numericBooleans) {
+    JsonNode doc = mapper.createObjectNode();
+    Map<String, Long> pathWriteTimes = new HashMap<>();
+    Map<String, List<JsonNode>> deadLeaves = new HashMap<>();
```

Comment:
Why do we convert deal leaves to json? It seems unnecessary since we do not return them to the client :thinking:

Reply:
What if we use custom classes to represent dead leaves? I suppose it should make the code more readable.

Reply:
That is true - custom class would be better - also re: the `Pair` above it should be represented as a class of its own probably instead of a Pair...I'll look into this

Reply:
Also dead leaves appear to be ignored in some call paths. Would it make sense to make the building of this map optional?

Reply:
+1 about using visitor interface to collect dead leaves.